### PR TITLE
scan init containers

### PR DESCRIPTION
**Pull request by Erezf-p**

### CODE REVIEW

File: pkg/scanner/scanner_test.go [hunk 1]
```diff
@@ -62,3 +62,37 @@ func Test_getImageHash(t *testing.T) {
...
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := getNormalizedImageNameWithTag(tt.args.image); got != tt.want {
+				t.Errorf("getNormalizedImageNameWithTag() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
```
Comment (FrimIdan) [inline pkg/scanner/scanner_test.go:98]: EOL

File: pkg/scanner/scanner.go [hunk 2]
```diff
@@ -69,15 +71,16 @@ type imagePodContext struct {
...
-	imageName  string
-	contexts   []*imagePodContext // All the pods that contain this image
-	scanUUID   string
-	result     []*clair.Vulnerability
-	resultChan chan bool
-	success    bool
-	completed  bool
-	timeout    bool
-	scanErrMsg string
+	imageName           string
```
Comment (FrimIdan) [inline pkg/scanner/scanner.go:74]: Is it being used?

**Review by FrimIdan — CHANGES_REQUESTED** [review]

File: pkg/scanner/scanner.go [hunk 3]
```diff
@@ -122,16 +123,19 @@ func (s *Scanner) initScan() error {
 		for _, container := range pod.Status.ContainerStatuses {
 			containerNameToImageId[container.Name] = container.ImageID
 		}
+		for _, container := range pod.Status.InitContainerStatuses {
+			containerNameToImageId[container.Name] = container.ImageID
+		}
 
-		for _, container := range pod.Spec.Containers {
+		_ = k8s_utils.VisitContainersWithPath(&pod.Spec, func(container *corev1.Container, _ *field.Path) error {
```
Comment (FrimIdan) [inline pkg/scanner/scanner.go:130]: You can create one slice from pod.Spec.Containers & pod.Spec.InitContainers and go over it - why the need to use `VisitContainersWithPath`?

File: go.mod [hunk 4]
```diff
@@ -14,6 +14,7 @@ require (
 	k8s.io/apimachinery v0.18.1
 	k8s.io/client-go v0.18.1
 	k8s.io/kubernetes v1.18.1
+	k8s.io/utils v0.0.0-20200324210504-a9aa75ae1b89
```
Comment (FrimIdan) [inline go.mod:17]: @Erezf-p do you still need this?

Comment (Erezf-p) [reply go.mod:17]: no. will remove

**Pull request merged**